sql/sql_connect.cc

```diff
@@ -84,6 +84,20 @@ static int increment_connection_count(THD* thd, bool use_lock);
 
 static HASH hash_user_connections;
 
+int get_user_con_by_name(const char *user) {
```

Comment:
You are right. We should account same user from different hosts as same user, thus please go back to the independent calculation. BTW, this also looks like a bug in Facebook patch, feel free to report to them too